### ISSUE TITLE
ZenDesk crash when attempting to access external resource zendesk.belvedere.Storage.grantPermissionsForUri

### ISSUE THREAD

**Issue by jkmassel** [woocommerce/woocommerce-android#871]
From Fabric:

```
Fatal Exception: java.lang.SecurityException: UID 10257 does not have permission to content://media/external/file/22020 [user 0]
       at android.os.Parcel.createException(Parcel.java:1969)
       at android.os.Parcel.readException(Parcel.java:1935)
       at android.os.Parcel.readException(Parcel.java:1885)
       at android.app.IActivityManager$Stub$Proxy.grantUriPermission(IActivityManager.java:4763)
       at android.app.ContextImpl.grantUriPermission(ContextImpl.java:1850)
       at android.content.ContextWrapper.grantUriPermission(ContextWrapper.java:781)
       at zendesk.belvedere.Storage.grantPermissionsForUri(Storage.java:55)
       at zendesk.belvedere.Belvedere.grantPermissionsForUri(Belvedere.java:246)
       at zendesk.belvedere.Belvedere.getViewIntent(Belvedere.java:131)
       at zendesk.support.request.CellBindHelper.openAttachment(CellBindHelper.java:95)
       at zendesk.support.request.CellBindHelper.access$000(CellBindHelper.java:32)
       at zendesk.support.request.CellBindHelper$3.onClick(CellBindHelper.java:85)
       at android.view.View.performClick(View.java:6612)
       at android.view.View.performClickInternal(View.java:6581)
       at android.view.View.access$3100(View.java:785)
       at android.view.View$PerformClick.run(View.java:25904)
       at android.os.Handler.handleCallback(Handler.java:873)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:280)
       at android.app.ActivityThread.main(ActivityThread.java:6710)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

There's a discussion of this issue at 
https://develop.zendesk.com/hc/en-us/community/posts/360004230468-Android-SDK-SE-crash-with-permissions

**Source:** 5c45b4b4f8b88c29630325a1-fabric

**Comment by AmandaRiu** [comment]
The original Zendesk ticket in the body of this issue was closed by Zendesk, but it looks like they've opened another one and were able to replicate the issue. According to the last comment: "We've fixed the issue, and we're planning a release soon."

 https://develop.zendesk.com/hc/en-us/community/posts/360004391688-Android-SDK-crash-with-permissions?input_string=Android%20Crash%20due%20to%20file%20URI%20permissions

Waiting for that release to be published to production.

**Comment by AmandaRiu** [comment]
Received an update on the Zendesk ticket:
```
The fix went out in version 2.3.0, and we're on 2.3.1 now, so I'd recommend upgrading to that.
```
We are currently running version 2.1.1. Will submit a PR to upgrade to 2.3.1.
